common/dynamicconfig/basic_client.go

```diff
@@ -0,0 +1,212 @@
+package dynamicconfig
```

Comment:
Feel free to review but FYI the code in this file was extracted from file_based_client.go. I.e. it's not new, just decoupled from the file-aware aspects of that client.